### PR TITLE
Reliability fixes and changes as suggested in jasp-test-release 

### PR DESCRIPTION
**Pull request by juliuspf**
fixes https://github.com/jasp-stats/jasp-test-release/issues/1382
fixes https://github.com/jasp-stats/jasp-test-release/issues/1383
fixes https://github.com/jasp-stats/jasp-test-release/issues/1395
fixes https://github.com/jasp-stats/jasp-test-release/issues/1396
fixes https://github.com/jasp-stats/jasp-test-release/issues/1397
fixes https://github.com/jasp-stats/jasp-test-release/issues/1398

### CODE REVIEW

File: R/uniDimCommon.R [hunk 1]
```diff
@@ -13,17 +13,36 @@
...
+               exitAnalysisIfErrors = TRUE)
+  } else {
+    .hasErrors(dataset = dataset,
+               type = c("infinity", "variance", "observations"),
+               observations.amount = " < 3",
+               infinity.target = options$variables,
+               variance.target = options$variables,
+               observations.target = options$variables,
+               exitAnalysisIfErrors = TRUE)
+  }
```
Comment (vandenman) [inline R/uniDimCommon.R:45]: you can unify these statements into something like this:

```r
.checkErrors <- function(dataset, options, Bayes = FALSE) {

  if (Bayes) {
    # check for sensible MCMC values
    custom <- list(checkMCMCBounds = function() {
      # are the remaining samples after burnin and thinning larger than 2?
      if (ceiling((options[["noSamples"]] - options[["noBurnin"]]) / options[["noThin"]]) < 2) {
        return(gettext("Too few MCMC samples will remain after the removal of the burnin samples and thinning,
                     please increase No.samples."))
      }
      return(NULL)
    })
  } else {
    custom <- NULL
  }

  .hasErrors(dataset = dataset,
             type = c("infinity", "variance", "observations"),
             observations.amount = " < 3",
             infinity.target = options$variables,
             variance.target = options$variables,
             observations.target = options$variables,
             custom = custom,
             exitAnalysisIfErrors = TRUE)
}
```
`.hasErrors` has an explicit argument for `custom` with default `NULL`.

**Review by vandenman — COMMENTED** [review]
code looks mostly fine to me.

File: R/uniDimCommon.R (hunk 1 above)
Comment (juliuspf) [reply R/uniDimCommon.R]: Do I need 


> ```r
>     custom <- list(checkMCMCBounds = function() {
> ```r
 the list argument here? It also works without it on my computer.

Comment (vandenman) [reply R/uniDimCommon.R]: you don't need that
```r
custom <- function() {
```
would also work. But as soon as you want to do multiple custom checks, they need to be put in a list.

**Comment by vandenman** [comment]
- [x] jasp-stats/jasp-test-release#1382 (with a CFA failed message)
- [x] jasp-stats/jasp-test-release#1383 (with a CFA failed message, the file `0.12.1ReliabilityError` took 30+ minutes to refresh...)
- [x] jasp-stats/jasp-test-release#1395
- [x] jasp-stats/jasp-test-release#1396
- [x] jasp-stats/jasp-test-release#1397
- [ ] jasp-stats/jasp-test-release#1398
I can still do this: 
image

there is a red box in QML, but the analysis still continues. I propose to either fix the the text in the table: 
image
 basically, sort the bounds, or alternatively fix the QML elements.

File: inst/qml/uniDimBayesian.qml [hunk 2]
```diff
@@ -231,6 +231,7 @@ Form
 				{
 					DoubleField
```
Comment (vandenman) [inline inst/qml/uniDimBayesian.qml:232]: In this DoubleField, you should set:
`max: probTableValueHigh.value`

where `probTableValueHigh` is the (new) id of the second DoubleField.

Comment (juliuspf) [reply inst/qml/uniDimBayesian.qml:232]: I have tried that, but it seems as though when the lower bound field is evaluated it does not see a higher value for the upper bound: 
Screenshot 2021-08-11 at 17 28 51

**Review by vandenman — APPROVED** [review]

File: R/uniDimBayesianOutputTables.R [hunk 3]
```diff
@@ -223,6 +233,7 @@
 
     df <- data.frame(statistic = opts[idxSelected], prior = probsPrior, posterior = probsPost)
     probTable$setData(df)
+    probTable$addFootnote(note)
```
Comment (vandenman) [inline R/uniDimBayesianOutputTables.R:236]: shouldn't you check here if `note != ""`?

Comment (juliuspf) [reply R/uniDimBayesianOutputTables.R:236]: Well. The footnote works when `note = ""`, there is just no note then. So why would I check it?

Comment (vandenman) [reply R/uniDimBayesianOutputTables.R:236]: > So why would I check it?

Because it shows that you don't intend to set a footnote to a future reader. Now, you depend on some undocumented implementation detail. 

Also, I'm not seeing anywhere in https://github.com/jasp-stats/jaspResults/blob/8807c66fc14fcdedb6576ee17c92af16a7e3fb48/R/zzzWrappers.R#L590-L603 or https://github.com/jasp-stats/jaspResults/blob/8807c66fc14fcdedb6576ee17c92af16a7e3fb48/src/jaspTable.cpp#L1100-L1117 where the case `""` is explicitly ignored, so this could also be a bug that is interpreted as a feature and that might change in the future.

Comment (juliuspf) [reply R/uniDimBayesianOutputTables.R:236]: Hmm okay. I was actually thinking that the object `note <- ""`is a character in R and thus it is fine to add it in a table footnote. I am doing this in several other instances, because I thought it was more handy than checking everytime if the footnote wasn't empty.
